pkg/resources/opniopensearch/resources.go

```diff
@@ -254,3 +282,49 @@ func calculateImage(repo, version, opensearchVersion string) string {
 		version,
 	)
 }
+
+func (r *Reconciler) pluginsList() []string {
+	plugins := []string{}
+	if r.instance.Spec.S3Settings != nil {
+		plugins = append(plugins, "repository-s3")
+	}
```

Comment:
Do we plan to also support the FileSystem snapshots in the future?

Reply:
It was pretty easy to implement so I have left the door open for that.  We may also support other storage providers besides S3

Reply:
Cool, I'll pick it up whenever I can.

Reply:
I created https://github.com/rancher/opni/issues/1676 so it doesn't get lost.